runtime/jilgen/jilconsts.c

```diff
@@ -752,6 +752,9 @@ writeConstants(OMRPortLibrary *OMRPORTLIB, IDATA fd)
 			writeConstant(OMRPORTLIB, fd, "J9TR_ELSSize", sizeof(J9VMEntryLocalStorage)) |
 			writeConstant(OMRPORTLIB, fd, "J9TR_J9_EXTENDED_RUNTIME_DEBUG_MODE", J9_EXTENDED_RUNTIME_DEBUG_MODE) |
 			writeConstant(OMRPORTLIB, fd, "J9TR_J9_EXTENDED_RUNTIME_USE_VECTOR_REGISTERS", J9_EXTENDED_RUNTIME_USE_VECTOR_REGISTERS) |
+#if defined(J9VM_OPT_VALHALLA_NESTMATES)
+			writeConstant(OMRPORTLIB, fd, "J9TR_J9_VTABLE_INDEX_DIRECT_METHOD_FLAG", J9_VTABLE_INDEX_DIRECT_METHOD_FLAG) |
```

Comment:
The JIT may require this to always be defined.  Either that, or the nestmates flag will need an ASM_ equivalent so they can ifdef the PicBuilder code.

Reply:
@jdmpapin can you give some insight on this?

Reply:
This `writeConstant()` allows us to write `J9TR_J9_VTABLE_INDEX_DIRECT_METHOD_FLAG` symbolically instead of hard-coding 1 in the assembly code in PicBuilder. I think what @gacholio is saying is that there is currently no `J9VM_OPT_VALHALLA_NESTMATES` flag to do conditional assembly in PicBuilder, so either we need to add such a flag, or else `J9TR_J9_VTABLE_INDEX_DIRECT_METHOD_FLAG` has to be unconditionally defined.

Reply:
Yes, that's exactly what I'm saying, so you guys can decide which path you prefer.